lxd/devices.go

```diff
@@ -823,3 +979,71 @@ func deviceParseDiskLimit(readSpeed string, writeSpeed string) (int64, int64, in
 
 	return readBps, readIops, writeBps, writeIops, nil
 }
+
+const USB_PATH = "/sys/bus/usb/devices"
+
+func loadRawValues(p string) (map[string]string, error) {
+	values := map[string]string{
+		"idVendor":  "",
+		"idProduct": "",
+		"dev":       "",
+		"busnum":    "",
+		"devnum":    "",
+	}
+
+	for k, _ := range values {
+		v, err := ioutil.ReadFile(path.Join(p, k))
+		if err != nil {
+			return nil, err
+		}
+
+		values[k] = strings.TrimSpace(string(v))
+	}
+
+	return values, nil
+}
+
+func deviceLoadUsb() ([]usbDevice, error) {
+	result := []usbDevice{}
+
+	ents, err := ioutil.ReadDir(USB_PATH)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
What happens on a system without USB? Does that only cause a failure when a usb device is added to the container or does that also affect containers without usb devices?